corehq/ex-submodules/casexml/apps/case/xform.py

```diff
@@ -264,6 +266,18 @@ def _validate_indices(case_db, cases):
                     invalid = True
                 if invalid:
                     # fail hard on invalid indices
+                    from distutils.version import LooseVersion
+                    if case_db.cached_xforms:
+                        xform = case_db.cached_xforms[0]
+                        if xform.metadata and xform.metadata.commcare_version:
+                            commcare_version = xform.metadata.commcare_version
+                            _soft_assert(
+                                commcare_version < LooseVersion("2.35"),
+                                "Invalid Case Index in CC version >= 2.35", {
+                                    'domain': case_db.domain,
+                                    'version': str(commcare_version)
```

Comment:
Could include the form id here like you are below, and maybe also the ID of the index that was invalid, in case that's helpful for investigating?